### PR TITLE
Stubs for Role Filter with LCE search

### PR DESCRIPTION
**Pull request by pgagne**
For https://bugzilla.redhat.com/show_bug.cgi?id=1569395

### CODE REVIEW

File: tests/foreman/api/test_role.py [hunk 1]
```diff
@@ -1792,3 +1792,57 @@ def test_negative_assign_org_admin_to_ldap_user_group(self):
...
         """
+
+
+class RoleSearchFilterTestCase(APITestCase):
+    """
+    Tests adding additional search filters to role filters.
+    """
+
+    @stubbed
+    @tier1
```
Comment (JacobCallahan) [inline tests/foreman/api/test_role.py]: These are likely T3

File: tests/foreman/api/test_role.py [hunk 2]
```diff
@@ -1792,3 +1792,57 @@ def test_negative_assign_org_admin_to_ldap_user_group(self):
...
+    def test_positive_role_lce_search(self):
+        """Test role with search filter using lifecycle enviroment.
+
+        :id: 7147f402-8cc0-4f6a-affe-5a0dd122a05a
+
+        :steps:
+            1. Created Lifecycle Environments DEV
```
Comment (JacobCallahan) [inline tests/foreman/api/test_role.py]: Can you use "Create" instead of "Created"?

File: tests/foreman/api/test_role.py [hunk 3]
```diff
@@ -1792,3 +1792,57 @@ def test_negative_assign_org_admin_to_ldap_user_group(self):
...
+            2. Created Host dev1 in DEV lifecycle environment.
+            3. Created role 'integrationsrole' role:
+                - see https://access.redhat.com/articles/3359731
```
Comment (JacobCallahan) [inline tests/foreman/api/test_role.py]: can you remove the reference you access.redhat.com?

File: tests/foreman/api/test_role.py [hunk 4]
```diff
@@ -1792,3 +1792,57 @@ def test_negative_assign_org_admin_to_ldap_user_group(self):
...
+                - Added search filter to view host rule "lifecycle_environment = DEV"
+                - Set Organization and location for integrationsrole to the same as dev1
+            4. Create apiuser with integrationsrole.
+            5. Using API with user apiuser read hosts facts of qa1.
+                (ie  curl -s -u 'apiuser:redhat' -k https://$SATELLITE/api/v2/hosts/$qa1_id/facts)
+
+        :expectedresults: The user is not allowed to view qa1 facts.
+
+        :BZ: 1651699
+        """
```
Comment (JacobCallahan) [inline tests/foreman/api/test_role.py]: largely the same comments for this one.

File: tests/foreman/api/test_role.py (hunk 3 above)
Comment (pgagne) [reply tests/foreman/api/test_role.py]: This is a public article. It's also a good reference which is why i included it.

**Pull request merged**